docs/src/topics/sparse_matrix.md

```diff
@@ -0,0 +1,207 @@
+(CSR)*](https://en.wikipedia.org/wiki/Sparse_matrix#Compressed_sparse_row_(CSR,_CRS_or_Yale_format))
+format or the [*compressed sparse column
+(CSC)*](https://en.wikipedia.org/wiki/Sparse_matrix#Compressed_sparse_column_(CSC_or_CCS))
+format, which is the default sparse matrix type implemented in the SparseArrays standard
+library. These matrix formats allow for fast linear algebra operations, such as
+factorizations and matrix-vector multiplications, that are needed when the linear system is
+solved. See [Instantiating the sparse matrix](@ref) for more details.
+
+In summary, a dynamic structure is more efficient when incrementally building the pattern by
+inserting new entries, and a static or compressed structure is more efficient for linear
+algebra operations.
+
+### Basic sparsity patterns construction in Ferrite
+
+Working with the sparsity pattern explicitly is in many cases not necessary. For basic
+usage (e.g. when only one matrix needed, when no customization of the pattern is
+required, etc) there exist convenience methods of [`create_matrix`](@ref) that return
+the matrix directly. For example, most examples in this documentation don't deal with
+the sparsity pattern explicitly.
+    
+### Costum sparsity patterns construction in Ferrite
+
+In more advanced cases, we need more fine grained control of the sparsity pattern. The 
+following steps are typically taken when constructing a sparsity pattern in Ferrite:
+
+ 1. *Initialize an empty pattern.* This can be done by either using the
+    [`init_sparsity_pattern(dh)`](@ref) function or by using a constructor directly.
+    `init_sparsity_pattern` will return a default pattern type that is compatible with the
+    DofHandler. In some cases you might require another type of pattern (for example a
+    blocked pattern) and in that case you can use the constructor directly.
+
+ 2. *Add entries to the pattern.* There are a number of functions that add entries to the
+    pattern:
+     - [`create_sparsity_pattern!`](@ref) adds entries for all couplings between the DoFs
+       within each element. These entries correspond to assembling the standard element
+       matrix and is thus almost always required.
+     - [`cross_element_coupling!`](@ref) adds entries for couplings between the DoFs in
+       neighboring elements. These entries are required when integrating along internal
+       interfaces between elements (e.g. for discontinuous Galerkin methods).
+     - [`Ferrite.add_entry!`](@ref) adds a single entry to the pattern. This can be used if
+       you need to add custom entries that are not covered by the other functions.
+
+ 3. *Condense the pattern.* This is done with the [`condense_sparsity_pattern!`](@ref)
+    function and adds entries required from constraints and boundary conditions in the
+    ConstraintHandler. Note that condensing the pattern *must* be done as the last operation
+    on the pattern, e.g. after adding all other entries.
```

Comment:
As discussed on Slack...
```suggestion
 2. **Add entries to the pattern:** This can conveniently be done by using [`create_sparsity_pattern!`](@ref). If a finer-grained control is required, there are a number of functions that add entries to the
    pattern:
     - [`cell_couplings!`](@ref) adds entries for all couplings between the DoFs
       within each element. These entries correspond to assembling the standard element
       matrix and is thus almost always required.
     - [`cross_element_coupling!`](@ref) adds entries for couplings between the DoFs in
       neighboring elements. These entries are required when integrating along internal
       interfaces between elements (e.g. for discontinuous Galerkin methods).
     - [`Ferrite.add_entry!`](@ref) adds a single entry to the pattern. This can be used if
       you need to add custom entries that are not covered by the other functions.
     - [`condense_sparsity_pattern!`](@ref) adds entries required from constraints and boundary conditions in the
    ConstraintHandler. Note that condensing the pattern *must* be done as the last operation
    on the pattern, e.g. after adding all other entries. 
```

Reply:
With the extra comments that I think `constraint_couplings!` is a better name than `condense_sparsity_pattern!`. 

Side-note: Sounds like having to call `condense_sparsity_pattern!` last can easily lead to user bugs. Would it make sense to add a `condensed::Ref{Bool}` (or `ScalarWrapper`) in `SparsityPattern` to allow the other methods error if trying to add entries after condensation?

Reply:
If you dont call the condense function last, I think you will at least get an error in the assembly-routine.

I think the name change make sense!

Reply:
> With the extra comments that I think `constraint_couplings!` is a better name than `condense_sparsity_pattern!`.
> 

The process is called condensation, right? So why should we use a different name for this?

> Side-note: Sounds like having to call `condense_sparsity_pattern!` last can easily lead to user bugs. Would it make sense to add a `condensed::Ref{Bool}` (or `ScalarWrapper`) in `SparsityPattern` to allow the other methods error if trying to add entries after condensation?

I think this is a really good idea.

Reply:
> The process is called condensation, right? So why should we use a different name for this?

Sorry, missed this comment @termi-official: My logic is that the condensation happens when you call `apply!`, but this step only adds additional entries due to the couplings introduced by the affine constraints.

Reply:
No problem. I see. I still think users reading `constraint_couplings` would expect something different happening. What about `add_constraint_pattern!` or even something like this?

Reply:
Yes, probably good to be consistent with using `add_XY!` or `XY!` here, and since 
`add_entry!` is kindof needed perhaps having add makes sense. For the `couplings` part, was just to be consistent with `cross_element_coupling!`. 
So the options could be
* `add_cell_pattern!`
* `add_cross_element_pattern!` (cell?)
* `add_entry!`
* `add_constraint_pattern!`
or 
* `add_cell_couplings!`
* `add_cross_element_couplings!` (cell?)
* `add_entry!`
* `add_constraint_couplings!`

I think I prefer `couplings`, as this is more logical for the others IMO.

Reply:
For me `pattern` makes sense too for the suffix since we insert some kind of pattern to the sparsity pattern.

What about `cross_element` -> `interface`?

Reply:
Perhaps
```julia
insert_cell_couplings!(...)
insert_interface_couplings!(...)
insert_constraint_couplings!(...)
insert_entry!(...)
```
or
```julia
create_cell_couplings!(...)
create_interface_couplings!(...)
create_constraint_couplings!(...)
create_coupling!(...)
```
or
```julia
add_cell_couplings!(...)
add_interface_couplings!(...)
add_constraint_couplings!(...)
add_coupling!(...)
```

Reply:
A typo I think, but any of the suggestions could work with `_coupling(s)` -> `entries` too.